.github/ISSUE_TEMPLATE/PROPOSAL_TEMPLATE.md

```diff
@@ -5,20 +5,36 @@ title: '[PROPOSAL]'
 labels: proposal
 assignees: ''
 ---
+## What/Why
```

Comment:
With the expanded detail a proposal written in this format would be more actionable.  Have proposals been harder to work with in the previous format and we are looking to get more details upfront?

If we accept this change, the larger template might stifle the writing of proposals because of the increased complexity.   How would you feel about enhancing our contributing guide [[link](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md)] to better set up proposals and maybe we reference that as a guide to make using this template more successful?

Reply:
> With the expanded detail a proposal written in this format would be more actionable. Have proposals been harder to work with in the previous format and we are looking to get more details upfront?

Anecdotally, i've heard that the proposal templates are sometimes difficult to understand when you're coming at them from the roadmap and don't have a lot of context. This brings additional details to the forefront to make it easier to understand what we are building and why, both so end-users can better understand upcoming features, and so our contributors have a better sense of what we are building, why, and how they might be able to contribute.


> How would you feel about enhancing our contributing guide [[link](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md)] to better set up proposals and maybe we reference that as a guide to make using this template more successful?

Yes! I think we can add "Proposals" right after "Feature requests". I think the intent here is that proposals should have a higher bar and be more actionable than feature requests, which can be a little less comprehensive. The intent here is to continue to keep feature requests light, and proposals are intended to be written by a developer who is proposing new functionality that they are intending to build. Even if some of the details are light, it helps us prioritize.

Reply:
I don't have too many more thoughts on the contents themselves - I've mentioned folks that have make proposals to see what they think about the changes and get more feedback, [[query](https://github.com/search?q=org%3Aopensearch-project+%22How+did+you+come+up+with+this+proposal%3F%22&type=Issues)].

> I'll draft a paragraph in the contributing guide

Look forward to it!